core/src/main/java/org/infinispan/globalstate/impl/GlobalConfigurationManagerImpl.java

```diff
@@ -98,10 +98,11 @@ void start() {
             EnumSet.of(InternalCacheRegistry.Flag.GLOBAL));
       parserRegistry = new ParserRegistry();
 
-      // Start the static caches first
-      Set<String> staticCacheNames = new HashSet<>(configurationManager.getDefinedCaches());
+      // Start the static caches first - use TreeSet so start order is consistent between runs (internal caches
+      // will normally be first as they start with _)
+      Set<String> staticCacheNames = new TreeSet<>(configurationManager.getDefinedCaches());
```

Comment:
This logic can break if the user ventures away from cache names in the format ` a-zA-Z0-9`. We can make this more robust by explicitly start the internal caches first and then user caches using a combination of `InternalCacheRegistry#filterPrivateCaches` and `InternalCacheRegistry#getInternalCacheNames`.

I think using a `TreeSet` with ^ is still beneficial though, as like you say it will make cache startup consistent between runs.

Reply:
Updated to explicitly start internal caches first.